crates/esplora/src/blocking_ext.rs

```diff
@@ -112,6 +164,235 @@ impl EsploraExt for esplora_client::BlockingClient {
+            let gap_limit_reached = if let Some(i) = last_active_index {
+                last_index >= i.saturating_add(stop_gap as u32)
+            } else {
+                last_index + 1 >= stop_gap as u32
+            };
+            if gap_limit_reached {
+                break;
+            }
+        }
+
+        Ok(last_active_index)
+    }
+
+    fn populate_using_spks<I: IntoIterator<Item = ScriptBuf>>(
+        &self,
+        tx_graph: &mut TxGraph<ConfirmationTimeHeightAnchor>,
+        spks: I,
+        parallel_requests: usize,
+    ) -> Result<(), Error>
+    where
+        I::IntoIter: ExactSizeIterator,
+    {
+        self.populate_using_keychain_spks(
+            tx_graph,
+            spks.into_iter().enumerate().map(|(i, spk)| (i as u32, spk)),
+            usize::MAX,
+            parallel_requests,
+        )
+        .map(|_| ())
+    }
+
+    fn populate_using_txids<I: IntoIterator<Item = Txid>>(
+        &self,
+        tx_graph: &mut TxGraph<ConfirmationTimeHeightAnchor>,
+        txids: I,
+        parallel_requests: usize,
+    ) -> Result<(), Error>
+    where
+        I::IntoIter: ExactSizeIterator,
+    {
+        enum EsploraResp {
+            TxStatus(TxStatus),
+            Tx(Option<Tx>),
+        }
+
+        let mut txids = txids.into_iter();
+        loop {
+            let handles = txids
+                .by_ref()
+                .take(parallel_requests)
+                .map(|txid| {
+                    let client = self.clone();
+                    let tx_already_exists = tx_graph.get_tx(txid).is_some();
+                    std::thread::spawn(move || {
+                        if tx_already_exists {
+                            client
+                                .get_tx_status(&txid)
+                                .map_err(Box::new)
+                                .map(|s| (txid, EsploraResp::TxStatus(s)))
+                        } else {
+                            client
+                                .get_tx_info(&txid)
+                                .map_err(Box::new)
+                                .map(|t| (txid, EsploraResp::Tx(t)))
+                        }
+                    })
+                })
+                .collect::<Vec<JoinHandle<Result<(Txid, EsploraResp), Error>>>>();
+
+            if handles.is_empty() {
+                break;
+            }
+
+            for handle in handles {
+                let (txid, resp) = handle.join().expect("thread must not panic")?;
+                match resp {
+                    EsploraResp::TxStatus(status) => {
+                        if let Some(anchor) = anchor_from_status(&status) {
+                            let _ = tx_graph.insert_anchor(txid, anchor);
+                        }
+                    }
+                    EsploraResp::Tx(Some(tx_info)) => {
+                        let _ = tx_graph.insert_tx(tx_info.to_tx());
```

Comment:
There was a discussion when the fee calculation for Electrum was implemented, that when we populate by Txids and Outpoints, we still want to add the prev TxOuts for fee calculation because there might be new relevant data. I just noticed that both the previous and new Esplora implementations appear to not actually do this, and was wondering if this needs to be added in?

Reply:
Added.